audio/Makefile

```diff
@@ -0,0 +1,158 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# To build the firmware you will need:
+#   SDCC from http://sdcc.sourceforge.net
+#   Git from https://git-scm.com/
+#
+# To build a firmware suitable for loading into RAM:
+#   make
+#
+# To build a firmware suitable for loading from EEPROM:
+#   make FLAGS="-DEEPROM"
+#   
+# To load a firmware:
+# 	make load
+# 	You will need HDMI2USB-mode-switch from
+# 	https://github.com/timvideos/HDMI2USB-mode-switch
+#
+# TODO: Use a common Makefile between all the firmware versions
+TARGET := audio
+FX2LIBDIR := ../third_party/fx2lib
+LINUXHEADERSDIR := ../third_party/fx2lib-linux-headers
+LIBS := $(FX2LIBDIR)/lib/fx2.lib
+INCS := -I sdcc -I$(FX2LIBDIR)/include -I$(LINUXHEADERSDIR) -I.
+
+# Settings specific for the TimVideo hdmi2usb firmware
+BOARD ?= fx2
+FLAGS +=-DDEBUG -DBOARD_$(BOARD)
+USE_16K := 1
+
+# Must only be hex numbers
+FIRMWARE_VERSION := $(shell date +%Y%m%d)
+
+CC_SRCS := descriptors.c firmware.c
+CC_OBJS := $(CC_SRCS:%.c=%.rel)
+
+CC := sdcc
+
+ifeq (${BOARD},atlys)
+VID := 1D50
+PID := 60B7
+DID := 0002
+else
+ifeq (${BOARD},fx2)
+VID := 1D50
+PID := 60B7
+DID := 0003
+else
+ifeq (${BOARD},opsis)
+VID := 2A19
+PID := 5442
+DID := 0002
+else
+$(error "Unknown board type '$(BOARD)'")
+endif
+endif
+endif
+
+CFLAGS += -DDATE=0x$(FIRMWARE_VERSION) -mmcs51 --std-c99 -DSDCC -Wa"-p" $(FLAGS)
+CFLAGS += -DDSCR_AREA=0x3e00 --code-size 0x3c00 --xram-size 0x0200
+CFLAGS += --xram-loc 0x3c00 -Wl"-b INT2JT=0x3f00"
+
+# Use make V=1 for a verbose build.
+ifndef V
+    Q_CC=@echo      '      CC ' $@;
+    Q_AS=@echo      '      AS ' $@;
+    Q_LINK=@echo    '    LINK ' $@;
+    Q_RM=@echo      '   CLEAN ';
+    Q_OBJCOPY=@echo ' OBJCOPY ' $@;
+    Q_GEN=@echo     '     GEN ' $@;
+endif
+
+.PHONY: all clean check check_int2jt
+
+all: $(TARGET).hex
+
+$(CC_SRCS) $(AS_SRCS): $(FX2LIBDIR)/lib/fx2.lib $(LINUXHEADERSDIR)/.git
+
+$(FX2LIBDIR)/lib/fx2.lib: $(FX2LIBDIR)/.git
+	cd $(dir $@) && make -j1
+
+# We depend on a file inside the directory as git creates an
+# empty dir for us.
+#
+# Note that although we have the variable FX2LIBDIR, the submodule
+# magic will always check it out in fx2lib/
+$(FX2LIBDIR)/.git $(LINUXHEADERSDIR)/.git: ../.gitmodules
+	git submodule sync --recursive -- $$(dirname $@)
+	git submodule update --recursive --init $$(dirname $@)
+	touch $@ -r ../.gitmodules
+
+$(TARGET).hex: $(CC_OBJS)
+	$(Q_LINK)$(CC) $(CFLAGS) -o $@ $+ $(LIBS)
+
+%.rel: %.c descriptors.h
+	$(Q_CC)$(CC) $(CFLAGS) -c --disable-warning 85 $(INCS) $<
+
+# FIXME: Replace this with the better version stuff the HDMI2USB-litex-firmware uses.
```

Comment:
I've changed this FIXME to instead replace this whole thing with stuff similar to the firmware which runs on the soft-cpu -- see https://github.com/timvideos/HDMI2USB-litex-firmware/blob/master/firmware/version_data.sh